cmd/imagec/docker.go

```diff
@@ -269,9 +257,35 @@ func FetchImageBlob(options ImageCOptions, image *ImageWithMeta) (string, error)
 
 	diffID = fmt.Sprintf("sha256:%x", diffIDSum.Sum(nil))
 
+	// this isn't an empty layer, so we need to calculate the size
 	if diffID != string(DigestSHA256EmptyTar) {
-		// bytesRead represents the size of the decompressed layer data
-		image.size = bytesRead
+		var layerSize int64
+
+		// seek to the beginning of the file
+		imageFile.Seek(0, 0)
+
+		// recreate the decompressed tar Reader
+		decompressedTar, err := archive.DecompressStream(imageFile)
+		if err != nil {
+			return "", err
+		}
+
+		// get a tar reader for access to the files in the archive
+		tr := tar.NewReader(decompressedTar)
+
+		// iterate through tar headers to get file sizes
+		for {
+			tarHeader, err := tr.Next()
+			if err == io.EOF {
+				break
+			}
+			if err != nil {
+				return "", err
+			}
+			layerSize += tarHeader.Size
+		}
+
+		image.size = layerSize
 	}
```

Comment:
Correct me if I'm wrong but this block uncompresses the tarball and keeps that in memory right?